awswrangler/lakeformation/_read.py

```diff
@@ -0,0 +1,192 @@
+"""Amazon Lake Formation Module gathering all read functions."""
+import concurrent.futures
+import itertools
+import logging
+from typing import Any, Dict, Iterator, List, Optional, Tuple, Union
+
+import boto3
+import pandas as pd
+from pyarrow import NativeFile, RecordBatchStreamReader
+
+from awswrangler import _utils, exceptions
+from awswrangler._config import apply_configs
+from awswrangler.lakeformation._utils import wait_query
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+def _execute_query(
+    query_id: str,
+    token_work_unit: Tuple[str, int],
+    boto3_session: Optional[boto3.Session] = None,
+) -> pd.DataFrame:
+    client_lakeformation: boto3.client = _utils.client(service_name="lakeformation", session=boto3_session)
+    token, work_unit = token_work_unit
+    messages: NativeFile = client_lakeformation.execute(QueryId=query_id, Token=token, WorkUnitId=work_unit)["Messages"]
+    return RecordBatchStreamReader(messages.read()).read_pandas()
+
+
+def _resolve_sql_query(
+    query_id: str,
+    chunked: Optional[bool] = None,
+    use_threads: bool = True,
+    boto3_session: Optional[boto3.Session] = None,
+) -> Union[pd.DataFrame, Iterator[pd.DataFrame]]:
+    client_lakeformation: boto3.client = _utils.client(service_name="lakeformation", session=boto3_session)
```

Comment:
I think it is better to create the possible session once at the beginning and then use it through all other underlining calls. Just to avoid the overhead to recreate a new session inside each called function.

Reply:
Agreed, passing session from original public function